src/lib/Commands/TestCommand/index.js

```diff
@@ -370,19 +403,42 @@ ${bootstrapCode}
 }
 
 // resolve modules
-__module_tests(__module_ImpUnit_Promise_exports, __module_impUnit_exports.ImpTestCase);
+__module_tests(__module_impUnit_exports.ImpTestCase);
 __module_tests_bootstrap(__module_impUnit_exports.ImpUnitRunner);
 
 ${reloadTrigger}
 `;
 
       agentCode =
         `#line 1 "${quoteFilename(agentLineControlFile)}"
-${(this._sourceCode.agent || '/* no agent source */')}
+${agentIncludeOrComment}
 `;
       // </editor-fold>
     }
 
+    agentCode = this._Builder.machine.execute(agentCode, {
+      __FILE__: testFile.name,
+      __PATH__: path.dirname(testFile.path)
+    });
+    deviceCode = this._Builder.machine.execute(deviceCode, {
+      __FILE__: testFile.name,
+      __PATH__: path.dirname(testFile.path)
+    });
+
+    if (this.debug) {
+      // create folder to dump preprocessed code
+      if (!fs.existsSync('./build')) {
+        fs.mkdirSync('./build');
+      }
+      // write dump preprocessed codes
+      fs.writeFile('./build/preprocessed.agent.nut', agentCode, (err) => {
```

Comment:
Please use the `preprocessed.<test file>` name to keep sources for all the tests, not just the last one executed.